Binary files phonon-4.5.0/.CMakeLists.txt.swp and phonon-4.5.0-new/.CMakeLists.txt.swp differ
diff --exclude=build -Nurd phonon-4.5.0/CMakeLists.txt phonon-4.5.0-new/CMakeLists.txt
--- phonon-4.5.0/CMakeLists.txt	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/CMakeLists.txt	2011-05-09 23:42:17.000000000 -0400
@@ -32,8 +32,18 @@
 set(PHONON_LIB_MAJOR_VERSION "4") # Only change on binary incompatible changes
 set(PHONON_LIB_MINOR_VERSION "5") # Only change on binary compatible changes with new interfaces
 set(PHONON_LIB_PATCH_VERSION "0") # Bump whenever you feel like it :P
+
+if (APPLE AND PHONON_USE_FRAMEWORKS)
+set(PHONON_LIB_VERSION "${PHONON_LIB_MAJOR_VERSION}")
+set(PHONON_LIB_SOVERSION "${PHONON_LIB_MAJOR_VERSION}.${PHONON_LIB_MINOR_VERSION}")
+set(PC_LIBS "-F\${libdir} -Wl,-framework,phonon")
+set(PC_CFLAGS "-F\${libdir} -I\${includedir}")
+else (APPLE AND PHONON_USE_FRAMEWORKS)
 set(PHONON_LIB_VERSION "${PHONON_LIB_MAJOR_VERSION}.${PHONON_LIB_MINOR_VERSION}.${PHONON_LIB_PATCH_VERSION}")
-set(PHONON_LIB_SOVERSION ${PHONON_LIB_MAJOR_VERSION})
+set(PHONON_LIB_SOVERSION "${PHONON_LIB_MAJOR_VERSION}")
+set(PC_LIBS "-L\${libdir} -lphonon")
+set(PC_CFLAGS "-I\${includedir}")
+endif (APPLE AND PHONON_USE_FRAMEWORKS)
 
 add_definitions(${QT_DEFINITIONS})
 remove_definitions(-DQT3_SUPPORT_WARNINGS -DQT3_SUPPORT)
diff --exclude=build -Nurd phonon-4.5.0/cmake/FindPhononInternal.cmake phonon-4.5.0-new/cmake/FindPhononInternal.cmake
--- phonon-4.5.0/cmake/FindPhononInternal.cmake	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/cmake/FindPhononInternal.cmake	2011-05-09 23:04:43.000000000 -0400
@@ -217,6 +217,7 @@
 # on the Mac support an extra install directory for application bundles
 if(APPLE)
     set(INSTALL_TARGETS_DEFAULT_ARGS ${INSTALL_TARGETS_DEFAULT_ARGS}
+                                     FRAMEWORK DESTINATION "${LIB_INSTALL_DIR}"
                                      BUNDLE DESTINATION "${BUNDLE_INSTALL_DIR}")
     set(CMAKE_SHARED_MODULE_CREATE_C_FLAGS   "${CMAKE_SHARED_MODULE_CREATE_C_FLAGS}   -flat_namespace -undefined dynamic_lookup")
     set(CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS "${CMAKE_SHARED_MODULE_CREATE_CXX_FLAGS} -flat_namespace -undefined dynamic_lookup")
Binary files phonon-4.5.0/gcc-root/g++-4.2 and phonon-4.5.0-new/gcc-root/g++-4.2 differ
Binary files phonon-4.5.0/gcc-root/gcc-4.2 and phonon-4.5.0-new/gcc-root/gcc-4.2 differ
diff --exclude=build -Nurd phonon-4.5.0/includes/CMakeLists.txt phonon-4.5.0-new/includes/CMakeLists.txt
--- phonon-4.5.0/includes/CMakeLists.txt	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/includes/CMakeLists.txt	2011-05-09 23:29:38.000000000 -0400
@@ -1,4 +1,4 @@
-install( FILES
+set(INCLUDES_HEADER_FILES
     phonon/AbstractAudioOutput
     phonon/AbstractMediaStream
     phonon/AbstractVideoOutput
@@ -50,8 +50,12 @@
     phonon/VolumeFaderEffect
     phonon/VolumeFaderInterface
     phonon/VolumeSlider
+)
+install( FILES ${INCLUDES_HEADER_FILES}
 DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
 
+install( FILES ${INCLUDES_HEADER_FILES}
+DESTINATION ${LIB_INSTALL_DIR}/phonon.framework/Versions/${PHONON_LIB_MAJOR_VERSION}/Headers COMPONENT Devel)
 
 install( FILES
   old/Phonon/AbstractAudioOutput
diff --exclude=build -Nurd phonon-4.5.0/phonon/CMakeLists.txt phonon-4.5.0-new/phonon/CMakeLists.txt
--- phonon-4.5.0/phonon/CMakeLists.txt	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/phonon/CMakeLists.txt	2011-05-09 23:30:08.000000000 -0400
@@ -109,6 +109,10 @@
   target_link_libraries(phonon m)
 endif (${CMAKE_SYSTEM_NAME} MATCHES "SunOS")
 
+if (APPLE AND PHONON_USE_FRAMEWORKS)
+set_target_properties(phonon PROPERTIES FRAMEWORK true)
+endif (APPLE AND PHONON_USE_FRAMEWORKS)
+
 set_target_properties(phonon PROPERTIES
                              VERSION ${PHONON_LIB_VERSION}
                              SOVERSION ${PHONON_LIB_SOVERSION}
@@ -117,7 +121,7 @@
 install(TARGETS phonon ${INSTALL_TARGETS_DEFAULT_ARGS})
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/phononnamespace.h.in ${CMAKE_CURRENT_BINARY_DIR}/phononnamespace.h)
 
-install(FILES
+set(PHONON_HEADER_FILES
    abstractaudiooutput.h
    abstractmediastream.h
    abstractvideooutput.h
@@ -155,6 +159,12 @@
    volumefaderinterface.h
    volumeslider.h
    ${CMAKE_CURRENT_BINARY_DIR}/phononnamespace.h
+)
+install(FILES
+   ${PHONON_HEADER_FILES}
    DESTINATION ${INCLUDE_INSTALL_DIR}/phonon COMPONENT Devel)
+install(FILES
+   ${PHONON_HEADER_FILES}
+   DESTINATION ${LIB_INSTALL_DIR}/phonon.framework/Versions/${PHONON_LIB_MAJOR_VERSION}/Headers COMPONENT Devel)
 
 install(FILES org.kde.Phonon.AudioOutput.xml  DESTINATION  ${DBUS_INTERFACES_INSTALL_DIR})
diff --exclude=build -Nurd phonon-4.5.0/phonon/experimental/CMakeLists.txt phonon-4.5.0-new/phonon/experimental/CMakeLists.txt
--- phonon-4.5.0/phonon/experimental/CMakeLists.txt	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/phonon/experimental/CMakeLists.txt	2011-05-09 23:06:17.000000000 -0400
@@ -23,6 +23,10 @@
    )
 automoc4_add_library(phononexperimental SHARED ${phononexperimental_LIB_SRCS})
 target_link_libraries(phononexperimental ${PHONON_LIBS})
+
+if (APPLE AND PHONON_USE_FRAMEWORKS)
+set_target_properties(phononexperimental PROPERTIES FRAMEWORK true)
+endif (APPLE AND PHONON_USE_FRAMEWORKS)
 set_target_properties(phononexperimental PROPERTIES
                                          VERSION ${PHONON_LIB_VERSION}
                                          SOVERSION ${PHONON_LIB_SOVERSION}
diff --exclude=build -Nurd phonon-4.5.0/phonon/experimental/tests/CMakeLists.txt phonon-4.5.0-new/phonon/experimental/tests/CMakeLists.txt
--- phonon-4.5.0/phonon/experimental/tests/CMakeLists.txt	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/phonon/experimental/tests/CMakeLists.txt	2011-05-09 22:40:07.000000000 -0400
@@ -28,22 +28,22 @@
 
 MACRO(PHONON_EXECUTABLE_TESTS)
    FOREACH(_testname ${ARGN})
-      set(_add_executable_param)
+      set(_phonon_add_executable_param)
       if (Q_WS_MAC)
-         set(_add_executable_param MACOSX_BUNDLE)
+         set(_phonon_add_executable_param MACOSX_BUNDLE)
       endif (Q_WS_MAC)
 # this breaks compilation      
 #      if (WIN32)
-#         set(_add_executable_param WIN32)
+#         set(_phonon_add_executable_param WIN32)
 #      endif (WIN32)
       if (NOT PHONON_BUILD_TESTS)
-         set(_add_executable_param ${_add_executable_param} EXCLUDE_FROM_ALL)
+         set(_phonon_add_executable_param ${_phonon_add_executable_param} EXCLUDE_FROM_ALL)
       endif (NOT PHONON_BUILD_TESTS)
       set(srcs ${_testname}.cpp)
       if (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${_testname}.qrc)
          qt4_add_resources(srcs ${_testname}.qrc)
       endif (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${_testname}.qrc)
-      automoc4_add_executable(${_testname} ${_add_executable_param} ${srcs})
+      automoc4_add_executable(${_testname} ${_phonon_add_executable_param} ${srcs})
       target_link_libraries(${_testname} ${PHONON_LIBS} phononexperimental ${QT_QTTEST_LIBRARY})
    ENDFOREACH(_testname)
 ENDMACRO(PHONON_EXECUTABLE_TESTS)
diff --exclude=build -Nurd phonon-4.5.0/phonon.pc.cmake phonon-4.5.0-new/phonon.pc.cmake
--- phonon-4.5.0/phonon.pc.cmake	2011-03-25 09:55:56.000000000 -0400
+++ phonon-4.5.0-new/phonon.pc.cmake	2011-05-09 23:14:26.000000000 -0400
@@ -8,5 +8,7 @@
 Description: Phonon library needed to build applications
 Version: @PHONON_LIB_MAJOR_VERSION@.@PHONON_LIB_MINOR_VERSION@.@PHONON_LIB_PATCH_VERSION@
 Requires: QtCore QtGui QtDBus
-Libs: -L${libdir} -lphonon
-Cflags: -I${includedir}
+Libs: @PC_LIBS@
+# -L${libdir} -lphonon
+Cflags: @PC_CFLAGS@
+# -I${includedir}
